### DISCUSSION TITLE
Virtual to Production

### DISCUSSION THREAD

**Discussion by Masterv98** [inventree/InvenTree#5191] [Ideas]
Hi, I'm the Brazilian translator (VGER), I'm also a 3Dprinting expert. So, I noticed that there is not a good flow for taking Virtual parts (like STEP, STL, OBJ, etc) to move directly to production for 3D print or some other digital way.

**Comment by wolflu05** [comment]
Hello @Masterv98,

you could attach your printing files as attachments to parts. There is no direct option to integrate your 3d printer with InvenTree, but you could write a [plugin](https://docs.inventree.org/en/latest/extend/how_to_plugin/) that is capable of sending your files to your 3d printers. This would actually also be a good usecase for the WIP 🚧 machine integration feature (#4824).

**Reply by Masterv98** [reply]
Ok got it. I'm not a good programmer at all. So my tip for anyone reading this is to integrate the Inventree with Octoprint (also an OSS). But as what I'm found is problems using virtual parts to make build parts. Maybe is a problem with my instance, not with the software itself.

**Reply by wolflu05** [reply]
I use octoprint to, really interesting idea. I could write it down on my list of projects, but it has absolute no priority to develop this (at least yet), because I have a lot of other stuff in my mind.

**Reply by Masterv98** [reply]
Im just happy I gave you this idea! Thanks
